batch-processing/README.md

```diff
@@ -0,0 +1,379 @@
+# Batch Processing
+
+In this project, you'll build a simple image processing pipeline: reading from a CSV file, downloading an image, processing it, and uploading it to cloud storage.
+
+> ⚠️ This project requires you to have access to an Amazon AWS account, with permissions to configure IAM and S3. Ask on CYF Slack for help with that.
+
+> ⚠️ You need a GitHub repo to complete this because we'll be using [GitHub Actions](https://docs.github.com/en/actions).
+
+Learning objectives:
+
+- What is batch processing and how does it differ from building servers?
+- How do we build resilience into batch processing software?
+- How do we use Go to use run existing software to complete tasks?
+- What is cloud storage technology and how do we read from it & upload data to it?
+<!-- * How do we deploy batch processing tasks in the cloud? -->
+
+Timebox: 2 days
+
+## Project
+
+This project starts with some scaffolding code already written. We'll be using a tool called [ImageMagick](https://imagemagick.org/) to process the images. The specific details of it are not important, so this is provided for us.
+
+ImageMagick is software written in C. We're going to call out to this code from Go using a package called [imagick](https://github.com/gographics/imagick), which acts an an interface between our Go code and the ImageMagick code.
+
+As a result of this dependency on ImageMagick, we're going to run this whole project in Docker. That means, rather than use `go run` or `go test`, we'll build and run a Docker container that has ImageMagick correctly installed.
+
+> It's possible to install ImageMagick on your computer to run this, but you don't _need_ to do that.
+
+The Docker commands get a bit complex so we can do things like run tests and _see_ the output of the code, so we're not going to type the Docker commands ourselves.
+
+Instead, to run the code, we use a tool called [Make](https://www.gnu.org/software/make/). Make enables us to build code without knowing the details of how that is done, because these details are recorded in the `Makefile`. In that way it's similar to Docker: the complexity of how to run the code in a container is supplied in the `Dockerfile`.
```

Comment:
```suggestion
Instead, to run the code, we use a tool called [Make](https://www.gnu.org/software/make/). Make enables us to build code without knowing the details of how that is done, because these details are recorded in the `Makefile`. In that way it's similar to Docker - when using Make you store the complexity of what to run in a `Makefile`, and when using Docker you store the complexity of how to run the code in a container in a `Dockerfile`.
```

I think there's quite a lot of nuance here; I'd maybe consider even just removing the comparison to Docker... Unsure...